File: app/src/main/java/rrzaniolo/testandroidsantander/network/models/ui/base/FieldCellView.java
package rrzaniolo.testandroidsantander.network.models.ui.base;

/*
 * Created by Rodrigo Rodrigues Zaniolo on 5/8/2018.
 * All rights reserved.
 */

import android.view.ViewGroup;

import rrzaniolo.testandroidsantander.network.models.data.FieldCell;

/***
 * Representation of Cell that requires some sort of Validation.
 * */
public abstract class FieldCellView extends CellView{

    //region --- Constructor
    public FieldCellView(FieldCell cell, ViewGroup viewGroup) {
        super(cell, viewGroup);
    }
    //endregion

    //region --- Methods
    public abstract Boolean isValidAnswer();

    public abstract void showError();
    public abstract void hideError();

    public abstract void clearField();
    //endregion
}

File: app/src/main/java/com/vctapps/santanderchallenge/form/presentation/domain/cell/base/FieldCellView.kt
package com.vctapps.santanderchallenge.form.presentation.domain.cell.base

import android.view.ViewGroup
import com.vctapps.santanderchallenge.form.domain.cell.FieldCell

abstract class FieldCellView(private val cell: FieldCell,
                             private val rootView: ViewGroup): CellView(cell, rootView){

    abstract fun isValidAnswer(): Boolean

    abstract fun showError()

    abstract fun hideError()

}